frontend/src/app/main/ui/components/editable_select.cljs

```diff
@@ -38,7 +41,21 @@
                               (let [value (-> event dom/get-target dom/get-value)
                                     value (or (d/parse-integer value) value)]
                                 (swap! state assoc :current-value value)
-                                (when on-change (on-change value))))]
+                                (when on-change (on-change value))))
+
+        on-node-load
+        (fn [node]
+          ;; There is a problem when changing the state in this callback that
+          ;; produces the dropdown to close in the same event
+          (timers/schedule
+           #(when-let [bounds (when node (dom/get-bounding-rect node))]
+              (let [{window-height :height} (dom/get-window-size)
+                    {:keys [left top height]} bounds
+                    bottom (when (< (- window-height top) 300) (- window-height top))
+                    top (when (>= (- window-height top) 300) (+ top height))]
+                (swap! state assoc :left left)
+                (swap! state assoc :top top)
+                (swap! state assoc :bottom bottom)))))]
```

Comment:
All this can be done in a single swap! i think.

Reply:
Yes, i'll change it